File: task-3/Go/Sub-Task 4.go
package main

import (
    "fmt"
    "io/ioutil"
    "strconv"
    "strings"
)

func main() {
    data, _ := ioutil.ReadFile("input.txt")
    n, _ := strconv.Atoi(strings.TrimSpace(string(data)))

    var output string
    for i := 0; i < n; i++ {
        output += strings.Repeat(" ", n-i-1) + strings.Repeat("*", 2*i+1) + "\n"
    }
    for i := n - 2; i >= 0; i-- {
        output += strings.Repeat(" ", n-i-1) + strings.Repeat("*", 2*i+1) + "\n"
    }

    ioutil.WriteFile("output.txt", []byte(output), 0644)
}

File: task-3/Rust/Sub-Task 4.rs
use std::fs;

fn main() {
    let data = fs::read_to_string("input.txt").expect("Unable to read file");
    let n: i32 = data.trim().parse().expect("Please enter a number");
    let mut output = String::new();

    for i in 0..n {
        output.push_str(&format!("{:width$}{}\n", "", "*".repeat((2 * i + 1) as usize), width = (n - i - 1) as usize));
    }
    for i in (0..n - 1).rev() {
        output.push_str(&format!("{:width$}{}\n", "", "*".repeat((2 * i + 1) as usize), width = (n - i - 1) as usize));
    }

    fs::write("output.txt", output).expect("Unable to write file");
}
